File: BotsSDK/korebot/src/main/java/kore/botssdk/audiocodes/webrtcclient/Fragments/TabletInfoFragment.java
package kore.botssdk.audiocodes.webrtcclient.Fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import kore.botssdk.R;
import kore.botssdk.audiocodes.webrtcclient.Callbacks.CallBackHandler;
import kore.botssdk.audiocodes.webrtcclient.General.ACManager;
import kore.botssdk.audiocodes.webrtcclient.General.Prefs;
import kore.botssdk.audiocodes.webrtcclient.Structure.SipAccount;

public class TabletInfoFragment extends BaseFragment implements FragmentLifecycle {

    private static final String TAG = "ChatFragment";

    private TextView stateValue;
    private TextView displayNameValue;
    private TextView userNameValue;


    private CallBackHandler.LoginStateChanged loginStateChanged = new CallBackHandler.LoginStateChanged() {
        @Override
        public void loginStateChange(boolean state) {
            handler.post(new Runnable() {
                @Override
                public void run() {
                    updateDate();
                }
            });
        }

    };



    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View rootView = inflater.inflate(R.layout.main_fragment_tablet_info, container, false);
        CallBackHandler.registerLginStateChange(loginStateChanged);
        initGui(rootView);
        return rootView;
    }

    private void initGui(View rootView)
    {
        stateValue = (TextView) rootView.findViewById(R.id.tablet_info_state_value_textview);
        displayNameValue = (TextView) rootView.findViewById(R.id.tablet_info_display_name_value_textview);
        userNameValue = (TextView) rootView.findViewById(R.id.tablet_info_user_name_value_textview);

        updateDate();
    }

    private void updateDate(){
        String statusStr = (ACManager.getInstance().isRegisterState() ? getString(R.string.account_textview_status_connected) : getString(R.string.account_textview_status_disconnected));
        stateValue.setText(statusStr);
        SipAccount sipAccount = Prefs.getSipAccount();
        displayNameValue.setText(sipAccount.getDisplayName());
        userNameValue.setText(sipAccount.getUsername());
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        CallBackHandler.unregisterLoginStateChange(loginStateChanged);
    }

    @Override
    public void onPauseFragment() {

    }

    @Override
    public void onResumeFragment() {

    }


}

File: BotSdk/ui/src/main/java/com/kore/ui/audiocodes/webrtcclient/fragments/TabletInfoFragment.kt
package com.kore.ui.audiocodes.webrtcclient.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import com.kore.ui.R
import com.kore.ui.audiocodes.webrtcclient.callbacks.CallBackHandler.LoginStateChanged
import com.kore.ui.audiocodes.webrtcclient.callbacks.CallBackHandler.registerLginStateChange
import com.kore.ui.audiocodes.webrtcclient.callbacks.CallBackHandler.unregisterLoginStateChange
import com.kore.ui.audiocodes.webrtcclient.general.Prefs
import com.kore.ui.databinding.MainFragmentTabletInfoBinding

class TabletInfoFragment : BaseFragment(), FragmentLifecycle {
    private var binding: MainFragmentTabletInfoBinding? = null
    private val loginStateChanged = LoginStateChanged { handler.post { updateUi() } }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = DataBindingUtil.inflate(layoutInflater, R.layout.main_fragment_tablet_info, container, false)
        registerLginStateChange(loginStateChanged)
        updateUi()
        return binding?.root
    }

    private fun updateUi() {
        val statusStr =
            if (acManager.isRegisterState) getString(R.string.account_textview_status_connected) else getString(R.string.account_textview_status_disconnected)
        binding?.apply {
            tabletInfoStateValueTextview.text = statusStr
            val sipAccount = Prefs.getSipAccount(requireContext())
            tabletInfoDisplayNameValueTextview.text = sipAccount.displayName
            tabletInfoUserNameValueTextview.text = sipAccount.username
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        binding = null
        unregisterLoginStateChange(loginStateChanged)
    }

    override fun onPauseFragment() {}
    override fun onResumeFragment() {}
}